app/src/features/pricing/components/PricingModal/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useCallback, useState, useEffect } from "react";
+import { Col, Modal, Row, Switch, Typography } from "antd";
+import { PricingTable, UpgradeWorkspaceMenu, PRICING } from "features/pricing";
+import { CompaniesSection } from "../CompaniesSection";
+import { CloseOutlined } from "@ant-design/icons";
+import { RQButton } from "lib/design-system/components";
+import { getFunctions, httpsCallable } from "firebase/functions";
+import { Checkout } from "./Checkout";
+import APP_CONSTANTS from "config/constants";
+import "./index.scss";
+
+interface PricingModalProps {
+  isOpen: boolean;
+  toggleModal: () => void;
+  selectedPlan?: string;
+  workspace?: any;
+}
+
+const PRIVATE_WORKSPACE = {
+  name: APP_CONSTANTS.TEAM_WORKSPACES.NAMES.PRIVATE_WORKSPACE,
+  id: "private_workspace",
+  accessCount: 1,
+};
+
+export const PricingModal: React.FC<PricingModalProps> = ({
+  isOpen,
+  toggleModal,
+  workspace = PRIVATE_WORKSPACE,
+  selectedPlan = null,
+}) => {
+  const [workspaceToUpgrade, setWorkspaceToUpgrade] = useState(workspace);
+  const [duration, setDuration] = useState(PRICING.DURATION.ANNUALLY);
+  const [stripeClientSecret, setStripeClientSecret] = useState(null);
+  const [isCheckoutScreenVisible, setIsCheckoutScreenVisible] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubscribe = useCallback(
+    (planName: string) => {
+      const functions = getFunctions();
+      const createTeamSubscriptionUsingStripeCheckout = httpsCallable(
+        functions,
+        "createTeamSubscriptionUsingStripeCheckout"
+      );
+      const createIndividualSubscriptionUsingStripeCheckout = httpsCallable(
+        functions,
+        "createIndividualSubscriptionUsingStripeCheckout"
+      );
+      setIsCheckoutScreenVisible(true);
+      setIsLoading(true);
+      if (workspaceToUpgrade?.id === PRIVATE_WORKSPACE.id) {
+        createIndividualSubscriptionUsingStripeCheckout({
+          currency: "usd",
+          teamId: null,
+          quantity: 1,
+          planName: planName,
+          duration: duration,
+        }).then((data: any) => {
+          setStripeClientSecret(data?.data?.payload.clientSecret);
+          setIsLoading(false);
+        });
+      } else {
+        createTeamSubscriptionUsingStripeCheckout({
+          currency: "usd",
+          teamId: workspaceToUpgrade?.id,
+          quantity: workspaceToUpgrade?.accessCount || 1,
+          planName: planName,
+          duration: duration,
+        }).then((data: any) => {
+          setStripeClientSecret(data?.data?.payload.clientSecret);
+          setIsLoading(false);
+        });
```

Comment:
Create the object passed to `createTeamSubscriptionUsingStripeCheckout` and then only call the function once instead of calling it conditionally
```suggestion 
  const subscriptionData = workspaceToUpgrade?.id === PRIVATE_WORKSPACE.id ?
          {..., quantity: 1,...} : {...quantity: workspaceToUpgrade?.accessCount || 1,...}
   createTeamSubscriptionUsingStripeCheckout(subscriptionData)
   .then((data: any) => setStripeClientSecret(data?.data?.payload.clientSecret))
   .finally(() =>  setIsLoading(false))
```

Reply:
There are 2 different function calls based on the workspace selected to upgrade one is `createIndividualSubscriptionUsingStripeCheckout` and other is `createTeamSubscriptionUsingStripeCheckout`

Reply:
Oh okay, but why we have two different functions with same intent which only differs by `quantity` ?